SPARQLWrapper/KeyCaseInsensitiveDict.py

```diff
@@ -2,47 +2,46 @@
 
 """
 A simple implementation of a key case-insensitive dictionary.
-
 ..
   Developers involved:
-
   * Ivan Herman <http://www.ivan-herman.net>
   * Sergio Fernández <http://www.wikier.org>
   * Carlos Tejo Alonso <http://www.dayures.net>
   * Alexey Zakhlestin <https://indeyets.ru/>
-
   Organizations involved:
-
   * `World Wide Web Consortium <http://www.w3.org>`_
   * `Foundation CTIC <http://www.fundacionctic.org/>`_
-
   :license: `W3C® Software notice and license <http://www.w3.org/Consortium/Legal/copyright-software>`_
 """
 
+from typing import Dict, TypeVar
+
+K = str
+V = TypeVar("V")
 
-class KeyCaseInsensitiveDict(dict):
+class KeyCaseInsensitiveDict(Dict[K, V]):
     """
     A simple implementation of a key case-insensitive dictionary
     """
 
-    def __init__(self, d={}):
+    def __init__(self, d: Dict[K, V]={}) -> None:
```

Comment:
This probably is of little consequence, but when typing it is best to use the broadest possible type, especially for arguments, in this case you only need something that implements [`collections.abc.Mapping`](https://docs.python.org/3/library/collections.abc.html#collections.abc.Mapping) - so best to use [`typing.Mapping`](https://docs.python.org/3/library/typing.html#typing.Mapping) here.

```suggestion
    def __init__(self, d: Mapping[K, V]={}) -> None:
```

It is fine as is though, up to you if you want to change it.

Reply:
485eaa6